orchagent/portsorch.cpp

```diff
@@ -886,46 +881,57 @@ bool PortsOrch::addVlanMember(Port vlan, Port port)
 {
     SWSS_LOG_ENTER();
 
-    sai_attribute_t attr;
-    vector<sai_attribute_t> attrs;
-
-    attr.id = SAI_VLAN_MEMBER_ATTR_VLAN_ID;
-    attr.value.u16 = vlan.m_vlan_id;
-    attrs.push_back(attr);
-
-    attr.id = SAI_VLAN_MEMBER_ATTR_BRIDGE_PORT_ID;
-    attr.value.oid = port.m_port_id;
-    attrs.push_back(attr);
-
+    sai_attribute_t vm_attrs[2];
+    vm_attrs[0].id = SAI_VLAN_MEMBER_ATTR_VLAN_ID;
+    vm_attrs[0].value.u16 = vlan.m_vlan_id;
+    vm_attrs[1].id = SAI_VLAN_MEMBER_ATTR_BRIDGE_PORT_ID;
+    vm_attrs[1].value.oid = port.m_port_id;
     sai_object_id_t vlan_member_id;
-    sai_status_t status = sai_vlan_api->create_vlan_member(&vlan_member_id, gSwitchId, attrs.size(), attrs.data());
-
+    sai_status_t status = sai_vlan_api->create_vlan_member(&vlan_member_id, gSwitchId, 2, vm_attrs);
     if (status != SAI_STATUS_SUCCESS)
     {
         SWSS_LOG_ERROR("Failed to add member %s to VLAN %s vid:%hu pid:%lx",
                 port.m_alias.c_str(), vlan.m_alias.c_str(), vlan.m_vlan_id, port.m_port_id);
         return false;
     }
-
     SWSS_LOG_NOTICE("Add member %s to VLAN %s vid:%hu pid%lx",
             port.m_alias.c_str(), vlan.m_alias.c_str(), vlan.m_vlan_id, port.m_port_id);
 
-    attr.id = SAI_PORT_ATTR_PORT_VLAN_ID;
-    attr.value.u16 = vlan.m_vlan_id;
-
-    status = sai_port_api->set_port_attribute(port.m_port_id, &attr);
+    // Set port bind mode
+    sai_attribute_t port_attr;
+    port_attr.id = SAI_PORT_ATTR_BIND_MODE;
+    port_attr.value.s32 = SAI_PORT_BIND_MODE_PORT;
+    status = sai_port_api->set_port_attribute(port.m_port_id, &port_attr);
     if (status != SAI_STATUS_SUCCESS)
     {
-        SWSS_LOG_ERROR("Failed to set port VLAN ID vid:%hu pid:%lx",
-                vlan.m_vlan_id, port.m_port_id);
+        SWSS_LOG_ERROR("Failed to set port pid:%lx bind mode to port: %d",
+                port.m_port_id, status);
         return false;
     }
 
-    SWSS_LOG_INFO("Set port %s VLAN ID to %hu", port.m_alias.c_str(), vlan.m_vlan_id);
+    // Create bridge ports and add ports to bridge
+    sai_object_id_t bport;
+    sai_attribute_t bport_attr[4];
+    bport_attr[0].id = SAI_BRIDGE_PORT_ATTR_TYPE;
+    bport_attr[0].value.s32 = SAI_BRIDGE_PORT_TYPE_PORT;
+    bport_attr[1].id = SAI_BRIDGE_PORT_ATTR_PORT_ID;
+    bport_attr[1].value.s32 = port.m_port_id;
+    bport_attr[2].id = SAI_BRIDGE_PORT_ATTR_BRIDGE_ID;
+    bport_attr[2].value.oid = m_default1QBridge;
+    bport_attr[3].id = SAI_BRIDGE_PORT_ATTR_FDB_LEARNING_MODE;
+    bport_attr[3].value.s32 = SAI_BRIDGE_PORT_FDB_LEARNING_MODE_HW;
+    status = sai_bridge_api->create_bridge_port(&bport, gSwitchId, 4, bport_attr);
```

Comment:
why is bridge port created here ?
for each port, there is a matching bridge port belonging to the .1q default bridge, on init
after reading the default .1q bridge, please read the bridge port list of it, then per bridge port read the port id, match to port obejct and add it this way
then, set the learning mode for the bridge port

the only time bridge port has to be created is when 
splitting port (we aren't there yet)
creating new lag (add the lag to bridge)
removing port from lag (add back the port to bridge)
i think we are also missing then the lag use cases code

so when adding a port to vlan, no need to create bridge port, just use the matching bridge port for the port